### PR TITLE
api(video): simplify video api

### PR DESCRIPTION
**Pull request by dgozman**
- This leaves just `recordVideos` and `videoSize` options on the context.
- Videos are saved to `artifactsPath`. We also save their file names to trace.
- `context.close()` waits for the processed videos.

### CODE REVIEW

File: src/server/browser.ts [hunk 1]
```diff
@@ -89,20 +87,18 @@ export abstract class Browser extends EventEmitter {
...
 
-  _videoStarted(videoId: string, file: string, pageOrError: Promise<Page | Error>) {
-    const video = new Video(file);
-    this._idToVideo.set(videoId, video);
+  _videoStarted(video: Video, pageOrError: Promise<Page | Error>) {
+    this._idToVideo.set(video._videoId, video);
     pageOrError.then(pageOrError => {
-      // Emit the event in another task to ensure that newPage response is handled before.
       if (pageOrError instanceof Page)
-        makeWaitForNextTask()(() => pageOrError.emit(Page.Events.VideoStarted, video));
```
Comment (yury-s) [inline src/server/browser.ts:98]: This was necessary for the following code to work:
```js
const page = await context.newPage();
const v = await page.waitForEvent('videostarted');
```
Do we not need this to work any more?

**Review by aslushnikov — APPROVED** [review]

File: src/server/firefox/ffPage.ts [hunk 2]
```diff
@@ -258,7 +257,8 @@ export class FFPage implements PageDelegate {
   }
 
   _onScreencastStarted(event: Protocol.Page.screencastStartedPayload) {
-    this._browserContext._browser._videoStarted(event.screencastId, event.file, this.pageOrError());
+    const video = new Video(this._browserContext, event.screencastId, event.file);
```
Comment (yury-s) [inline src/server/firefox/ffPage.ts:260]: Let's revert this change, the video creation is now repeated 3 times for no good reason.

**Review by yury-s — APPROVED** [review]
The API where you cannot attribute video to its page is somewhat weird but as discussed offline let's keep it this way until we have a compelling use case.

File: src/server/browser.ts (hunk 1 above)
Comment (dgozman) [reply src/server/browser.ts:98]: Yeah, we don't have an event anymore, so this is not a problem currently.

File: src/server/firefox/ffPage.ts (hunk 2 above)
Comment (dgozman) [reply src/server/firefox/ffPage.ts]: Done.
